docs/eventsV2.md

```diff
@@ -0,0 +1,268 @@
+# Queues V2 (updated May 8, 2022)
+
+## Design of Dark Queues V2
+
+Goals:
+
+- do not do DB queries to find events to run (that is, move scheduling outside DB)
+- support existing queue features (paused/blocked, delayed retries)
+- do not run a scheduling service
+
+## Definitions
+
+- `emit` adding an event to the queue from a Cron handler or the `emit` function
+- `event`: refers to the Dark value `emit`ed into the queue and the metadata around
+  it (including what you might call a "job" or "message" in other systems). We say we
+  "run the event" when we execute it, or "retry the event" if we requeue it.
+- `QueueWorker`: one of the machines which is running 1 or more events.
```

Comment:
may be worth mentioning that this may refer to one of:
- the the QueueWorker project/code
- the actual machine running the events